rs/test_utilities/state/src/lib.rs

```diff
@@ -767,15 +767,14 @@ pub fn register_callback(
     canister_state: &mut CanisterState,
     originator: CanisterId,
     respondent: CanisterId,
-    callback_id: CallbackId,
     deadline: CoarseTime,
-) {
+) -> CallbackId {
     let call_context_manager = canister_state
         .system_state
         .call_context_manager_mut()
         .unwrap();
     let call_context_id = call_context_manager.new_call_context(
-        CallOrigin::CanisterUpdate(originator, callback_id, deadline),
+        CallOrigin::SystemTask,
```

Comment:
Does it matter that it's now a `SystemTask` instead of a `CanisterUpdate`?

Reply:
Not from what I can tell. `SystemTask` is a heartbeat or a timer.  The [sanity check](https://sourcegraph.com/github.com/dfinity/ic@rumenov/regtypes/-/blob/rs/replicated_state/src/canister_state/system_state/call_context_manager.rs?L599) we want to pass doesn't care about the call origin.